### PR TITLE
Integration with Azure

### PR DESCRIPTION
**Pull request by kuraleta**

### CODE REVIEW

File: langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.model.openai.InternalOpenAiHelper.aiMessageFrom;
+import static dev.langchain4j.model.openai.InternalOpenAiHelper.toFunctions;
+import static dev.langchain4j.model.openai.InternalOpenAiHelper.toOpenAiMessages;
+import static java.time.Duration.ofSeconds;
+import static java.util.Collections.singletonList;
+
+/**
+ * Represents a chat model backed by Azure OpenAI.
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java]: Represents a connection to the OpenAI LLM, hosted on Azure, that has a chat completion interface (like gpt-3.5-turbo and gpt-4).

File: langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiLanguageModel.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import java.net.Proxy;
+import java.time.Duration;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static java.time.Duration.ofSeconds;
+
+/**
+ * Represents a connection to the Azure OpenAI LLM with a completion interface.
+ * However, it's recommended to use {@link OpenAiChatModel} instead,
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiLanguageModel.java]: OpenAiChatModel -> AzureOpenAiChatModel

File: langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiEmbeddingModel.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+import static java.time.Duration.ofSeconds;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Represents a connection to the OpenAI embedding model, such as text-embedding-ada-002, accessing it via Azure.
+ * <p>
+ * <a href="https://learn.microsoft.com/en-us/azure/ai-services/openai/reference">More information</a>
+ * <p>
+ */
+public class AzureOpenAiEmbeddingModel implements EmbeddingModel, TokenCountEstimator {
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiEmbeddingModel.java:36]: please add the same comment about authentication methods

File: langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiEmbeddingModel.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import java.time.Duration;
+import java.util.List;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static java.time.Duration.ofSeconds;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Represents a connection to the OpenAI embedding model, such as text-embedding-ada-002, accessing it via Azure.
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiEmbeddingModel.java]: Represents a connection to the OpenAI embedding model, hosted on Azure (like text-embedding-ada-002).

File: langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiLanguageModel.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+import java.net.Proxy;
+import java.time.Duration;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static java.time.Duration.ofSeconds;
+
+/**
+ * Represents a connection to the Azure OpenAI LLM with a completion interface.
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiLanguageModel.java]: Represents a connection to the OpenAI LLM, hosted on Azure (like text-davinci-003).

File: langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiStreamingChatModel.java [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+import java.util.List;
+
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.model.openai.InternalOpenAiHelper.toFunctions;
+import static dev.langchain4j.model.openai.InternalOpenAiHelper.toOpenAiMessages;
+import static java.time.Duration.ofSeconds;
+import static java.util.Collections.singletonList;
+
+/**
+ * Represents a connection to the Azure OpenAI with a chat completion interface.
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiStreamingChatModel.java]: Represents a connection to the OpenAI LLM, hosted on Azure, that has a chat completion interface (like gpt-3.5-turbo and gpt-4).

File: langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiLanguageModel.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                                    Tokenizer tokenizer,
+                                    Double temperature,
+                                    Duration timeout,
+                                    Integer maxRetries,
+                                    Proxy proxy,
+                                    Boolean logRequests,
+                                    Boolean logResponses) {
+
+        temperature = temperature == null ? 0.7 : temperature;
+        timeout = timeout == null ? ofSeconds(15) : timeout;
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiLanguageModel.java]: please make it 60 sec

File: langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiStreamingChatModel.java [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+                                         Integer maxTokens,
+                                         Double presencePenalty,
+                                         Double frequencyPenalty,
+                                         Duration timeout,
+                                         Proxy proxy,
+                                         Boolean logRequests,
+                                         Boolean logResponses) {
+
+        temperature = temperature == null ? 0.7 : temperature;
+        timeout = timeout == null ? ofSeconds(5) : timeout;
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiStreamingChatModel.java]: please make it 15 secs

File: langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiStreamingLanguageModel.java [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+import lombok.Builder;
+
+import java.net.Proxy;
+import java.time.Duration;
+
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static java.time.Duration.ofSeconds;
+
+/**
+ * Represents a connection to the OpenAI LLM with a completion interface, such as text-davinci-003.
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiStreamingLanguageModel.java]: Represents a connection to the OpenAI LLM, hosted on Azure (like text-davinci-003).

File: langchain4j/src/main/java/dev/langchain4j/model/openai/OpenAiLanguageModel.java [hunk 10]
```diff
@@ -38,13 +40,15 @@ public OpenAiLanguageModel(String apiKey,
                                Boolean logRequests,
                                Boolean logResponses) {
 
+        baseUrl = baseUrl == null ? OPENAI_URL : baseUrl;
         modelName = modelName == null ? TEXT_DAVINCI_003 : modelName;
         temperature = temperature == null ? 0.7 : temperature;
         timeout = timeout == null ? ofSeconds(15) : timeout;
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/model/openai/OpenAiLanguageModel.java:46]: could you please change timeout to 60 sec?

File: langchain4j-parent/pom.xml [hunk 11]
```diff
@@ -281,7 +281,9 @@
                             <scopes>compile,runtime,provided,test</scopes>
                             <acceptableLicenses>
                                 <license>
-                                    <name>(Apache License, Version 2\.0)|(Apache-2\.0)|(The Apache Software License, Version 2\.0)</name>
+                                    <name>(Apache License, Version 2\.0)|(Apache-2\.0)|(The Apache Software License,
+                                        Version 2\.0)
```
Comment (langchain4j) [inline langchain4j-parent/pom.xml]: please revert

**Review by ghost — APPROVED** [review]

**Pull request merged**